addons/markinglaser/functions/fnc_renderPFH.sqf

```diff
@@ -15,80 +15,53 @@
+    TRACE_1("end PFEH",GVAR(pfEH));
+};
 
 {
-    private _aircraft = _x;
-    private _gimbalLimits = _aircraft getVariable [QGVAR(gimbalLimits), []];
+    _y params ["_aircraft", "", "_laserMethod"];
 
-    // Get laser direction vector
-    private _vector = if (_aircraft getVariable QGVAR(useTurret)) then {
-        [_aircraft] call FUNC(getTurretVector)
-    } else {
-        [_aircraft] call FUNC(getPilotCamVector)
-    };
+    private _currentMode = _aircraft getVariable [QGVAR(laserMode), MODE_OFF];
+    if (_currentMode == MODE_OFF) then { continue };
+    if (_laserMethod != QEFUNC(laser,findLaserSource)) then { continue }; // Normal vanilla laserTarget func
 
-    // Turn off laser if gimbal limit is reached
-    // 0.1 is added to calculated values, seems to be slightly inaccurate
-    _gimbalLimits params ["_minDir", "_maxDir", "_minElev", "_maxElev"];
-    private _modelVector = _aircraft vectorWorldToModelVisual _vector;
-    private _dir = _modelVector # 0 atan2 _modelVector # 1;
-    if (_dir < _minDir - 0.1 || {_dir > _maxDir + 0.1}) then {
-        continue;
-    };
+    if ((_currentMode == MODE_FLASH) && {
+        private _cycle = (CBA_missionTime + (_aircraft getVariable [QGVAR(flashOffset), 0])) % (1/3);
+        _cycle > 1/6
+    }) then { continue };
 
-    private _elevation = _modelVector # 2 atan2 vectorMagnitude [_modelVector # 0, _modelVector # 1, 0];
-    if (_elevation < _minElev - 0.1 || {_elevation > _maxElev + 0.1}) then {
-        continue;
-    };
+    private _targetObject = _aircraft getVariable [QEGVAR(laser,targetObject), objNull];
+    private _targetPosASL = getPosASL _targetObject;
 
-    private _origin = [0, 0, 0];
-    if (_aircraft == cameraOn && {cameraView == "GUNNER"}) then {
-        _origin = AGLToASL positionCameraToWorld [0, 0, 0];
-    } else {
-        private _originPoint = _aircraft getVariable [QGVAR(laserOrigin), ""];
-        _origin = _aircraft modelToWorldVisualWorld (_aircraft selectionPosition _originPoint);
-    };
+    (_y call EFUNC(laser,findLaserSource)) params ["_laserPosASL", "_laserDir"];
 
-    // Check if laser hits object or ground
-    private _startPos = _origin vectorAdd (_vector vectorMultiply 0.25);
-    private _endPos = _startPos vectorAdd (_vector vectorMultiply LASER_MAX);
+    #ifdef DEBUG_MODE_FULL
+    drawIcon3D ["\a3\ui_f\data\IGUI\Cfg\Cursors\select_target_ca.paa", [1,0,1,1], (ASLToAGL _targetPosASL), 0.5, 0.5, 0, "Laser", 0.5, 0.025, "TahomaB"];
+    drawIcon3D ["\a3\ui_f\data\IGUI\Cfg\Cursors\select_target_ca.paa", [1,0,1,1], (ASLToAGL _laserPosASL), 0.5, 0.5, 0, "Origin", 0.5, 0.025, "TahomaB"];
+    #endif
+
+    private _smoothing = _aircraft getVariable [QGVAR(smoothing), []];
+    _smoothing pushBack _laserDir;
+    if (count _smoothing > 5) then { _smoothing deleteAt 0 };
+    private _smoothDir = [0,0,0];
+    { _smoothDir = _smoothDir vectorAdd _x } forEach _smoothing;
+    _smoothDir = _smoothDir vectorMultiply (1/count _smoothing);
```

Comment:
I know absolutely nothing about `drawLaser`, so please forgive my ignorance, but what was wrong with the old method of drawing it to max distance and adding onto it from there?